### PR TITLE
Adjust Gatekeeper Policy Installation For Improved CLI Installations

### PR DESCRIPTION
**Pull request by cartyc**
This PR moves the gatekeeper `constraints` from the `gatekeeper-policies` package to the `core-landing-zone` package. This will help resolve issue #409 and allow for easier non-gitops based installation approaches as this allows for the `ConstraintTemplate` to be installed before the gatekeeper `constraint` file and prevents the issue where one would need to comment out of the `constraint` file first before apply the `gatekeeper-policies` package.

The overall goal of this PR is to improve the installation process for users of non-gitops based installs, `kpt` in particular. This would also allow us to install more `constraintTemplates` via the `gatekeeper-policies` package into the cluster without needing to enable them right away.

### CODE REVIEW

**Review by fmichaelobrien — APPROVED** [review]
LGTM
- nice, thanks for the constraints move - Chris

**Comment by cartyc** [comment]
@alaincormier-ssc @davelanglois-ssc @fmichaelobrien I made the changes discussed yesterday and wanted to get your input before I start fixing the README and Kptfiles.

Created a new solution directory called `policy-constraint-bundles` with 2 bundles, `guardrails-v1` and `naming-policies`.

**Comment by alaincormier-ssc** [comment]
> @alaincormier-ssc @davelanglois-ssc @fmichaelobrien I made the changes discussed yesterday and wanted to get your input before I start fixing the README and Kptfiles.
> 
> Created a new solution directory called `policy-constraint-bundles` with 2 bundles, `guardrails-v1` and `naming-policies`.

that should work
will each bundle be tagged by release-please?

**Comment by cartyc** [comment]
That's what I was thinking, might be overkill but I believe having specific task based bundles makes sense. I was using the Anthos [Gatekeeper Bundles](https://github.com/GoogleCloudPlatform/acm-policy-controller-library/tree/main/bundles) as by point of reference.

**Comment by davelanglois-ssc** [comment]
Hey @cartyc, I really like the concept. If I understand correctly, the constrainttemplate would remain in the existing `gatekeeper-policies` package. Do you think that name still make sense ?

**Comment by cartyc** [comment]
> Hey @cartyc, I really like the concept. If I understand correctly, the constrainttemplate would remain in the existing `gatekeeper-policies` package. Do you think that name still make sense ?

That's correct, the package would just act as a means to load the `constrainttemplates` into the Config Controller or GKE instances, similar to what happens when you you enable the installation of the default template library on policy controller I think the name `gatekeeper-policies` still works but maybe the bundle names might need some additional tuning to help keep things more distinct. `gatekeeper-policy-templates` might be a good option though if we want to rename the package.

**Review by fmichaelobrien — DISMISSED** [review]
LGTM
- nice
- dryrun is appreciated - in particular for the cost-center/centre ones
- will reapprove if required after lint commit

**Comment by cartyc** [comment]
Thanks Mike, I was just cracking open an editor to start hacking away at the linting issues.

**Review by fmichaelobrien — DISMISSED** [review]
LGTM

**Review by fmichaelobrien — DISMISSED** [review]
LGTM
reviewed
https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/commit/c62264b1890434f1b1668d1c997834919f62863f#diff-0e4828475a445061adf098b63da0946c315796e05bb5354d1b142e2ec8b6d427

**Review by fmichaelobrien — DISMISSED** [review]
LGTM

**Review by fmichaelobrien — APPROVED** [review]
LGTM
- yes lint is soooo particular on CR/LF spaces...
- thanks for taking the heat

File: solutions/policy-constraint-bundles/guardrails-v1/Kptfile
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/guardrails-v1/Kptfile:1]: I suggest renaming this folder from policy-constraint-bundles to gatekeeper-policy-constraint-bundles

File: solutions/policy-constraint-bundles/guardrails-v1/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+### How this Policy Helps
+
+Validates no account has been created with the required permissions to access the MarketPlace `roles/cloudprivatecatalogproducer.admin`
+
+## Usage
+
+### Fetch the package
+
+`kpt pkg get https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit.git/solutions/policy-constraint-bundles/guardrails-v1 guardrails-v1`
```
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/guardrails-v1/README.md:108]: you would need to update this link if you decide to change the folder name

File: solutions/policy-constraint-bundles/std-protectedb-policies/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```yaml
+client_code: ^(aa|bb|cc)
+env_code: d # corresponds to environment `d` 
+region_code: m # Region Code `m` for montreal
+class_code: (u|a|b)- # unclassified | protected a | protected b
+user_defined_string: '^[a-z][a-z0-9-]*[a-z0-9]$' # user string
+```
+
+### Required Labels
+All Projects must have be labeled with a cost center code that matches the following pattern `^[a-zA-Z]+$`
```
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/std-protectedb-policies/README.md:40]: All projects must be labeled with a cloud business requirements ID that is represented with 4 digits

Below is the guidance. All cloud service provider environment should be using this standard tag:​

           TagName = ssc_cbrid ​

           Value = 4digit CBR ID

File: solutions/policy-constraint-bundles/std-protectedb-policies/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Usage
+
+### Fetch the package
+`kpt pkg get https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit.git/solutions/policy-constraint-bundles/std-protectedb-policies protectb-policies`
```
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/std-protectedb-policies/README.md:45]: update this link if you change the  folder name under solutions/

File: solutions/policy-constraint-bundles/std-protectedb-policies/README.md (hunk 2 above)
Comment (davelanglois-ssc) [reply solutions/policy-constraint-bundles/std-protectedb-policies/README.md:40]: this requirement is mandatory to all environements including profile 1, 2, 3, 4, 5, 6

File: solutions/policy-constraint-bundles/std-protectedb-policies/required-labels/cost-center.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+spec:
+  enforcementAction: dryrun
+  match:
+    kinds:
+      - apiGroups:
+          - resourcemanager.cnrm.cloud.google.com
+        kinds:
+          - Project
+  parameters:
+    message: "All namespaces must have an `cost-center` label"
```
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/std-protectedb-policies/required-labels/cost-center.yaml:14]: to be adjusted as per comment on README.md

File: solutions/policy-constraint-bundles/std-unclass-policies/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+```shell
+client_code: ^(aa|bb|cc)
+env_code: d # corresponds to environment `d` 
+region_code: m # Region Code `m` for montreal
+class_code: (u|a|b)- # unclassified | protected a | protected b
+user_defined_string: '^[a-z][a-z0-9-]*[a-z0-9]$' # user string
+```
+
+### Required Labels
+All Projects must have be labeled with a cost center code that matches the following pattern `^[a-zA-Z]+$`
```
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/std-unclass-policies/README.md:41]: to be adjusted as per comment on std-protectedb-policies/README.md

File: solutions/policy-constraint-bundles/std-unclass-policies/README.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+## Usage
+
+### Fetch the package
+```shell
+kpt pkg get https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit.git/solutions/policy-constraint-bundles/std-protectedb-policies unclass-policies
```
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/std-unclass-policies/README.md:48]: update this url as per folder name under solutions/

File: solutions/policy-constraint-bundles/std-unclass-policies/required-labels/cost-center.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+spec:
+  enforcementAction: dryrun
+  match:
+    kinds:
+      - apiGroups:
+          - resourcemanager.cnrm.cloud.google.com
+        kinds:
+          - Project
+  parameters:
+    message: "All namespaces must have an `cost-center` label"
```
Comment (davelanglois-ssc) [inline solutions/policy-constraint-bundles/std-unclass-policies/required-labels/cost-center.yaml:14]: to be adjusted as per comment on std-protectedb-policies/README.md

**Review by davelanglois-ssc — CHANGES_REQUESTED** [review]

File: solutions/policy-constraint-bundles/guardrails-v1/Kptfile
Comment (cartyc) [reply solutions/policy-constraint-bundles/guardrails-v1/Kptfile:1]: Renamed, like the added clarity. Update the links in the constraint packages as well

**Comment by cartyc** [comment]
@davelanglois-ssc should be good for a final review.

**Review by fmichaelobrien — DISMISSED** [review]
LGTM
the dryrun is appreciated

File: solutions/gatekeeper-policy-constraint-bundles/std-unclass-policies/required-labels/cost-center.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: K8sRequiredLabels
+metadata:
+  name: must-have-cost-center
```
Comment (davelanglois-ssc) [inline solutions/gatekeeper-policy-constraint-bundles/std-unclass-policies/required-labels/cost-center.yaml:4]: should be renamed to must-have-cbrid

File: solutions/gatekeeper-policy-constraint-bundles/guardrails-v1/required-labels/cost-center.yaml [hunk 9]
```diff
@@ -11,7 +11,7 @@ spec:
         kinds:
           - Project
   parameters:
-    message: "All namespaces must have an `cost-center` label"
+    message: "All Projects must have an `ssc_cbrid` label with a 4digit CBR ID"
```
Comment (davelanglois-ssc) [inline solutions/gatekeeper-policy-constraint-bundles/guardrails-v1/required-labels/cost-center.yaml:14]: minor, missing space between 4 and digit

**Review by fmichaelobrien — DISMISSED** [review]
LGTM
- last change good

File: solutions/gatekeeper-policy-constraint-bundles/std-unclass-policies/required-labels/cost-center.yaml (hunk 8 above)
Comment (cartyc) [reply solutions/gatekeeper-policy-constraint-bundles/std-unclass-policies/required-labels/cost-center.yaml]: Updated constraint name, thanks Dave!

**Review by davelanglois-ssc — APPROVED** [review]

**Review by fmichaelobrien — APPROVED** [review]
LGTM

**Comment by obriensystems** [comment]
closing - keep reference code only

**Pull request closed without merging**